Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/testresult/negative/SubmissionTestResultNegativeViewModel.kt

```diff
@@ -0,0 +1,62 @@
+package de.rki.coronawarnapp.ui.submission.testresult.negative
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.asLiveData
+import androidx.navigation.NavDirections
+import com.squareup.inject.assisted.AssistedInject
+import de.rki.coronawarnapp.storage.LocalData
+import de.rki.coronawarnapp.storage.SubmissionRepository
+import de.rki.coronawarnapp.ui.submission.testresult.TestResultUIState
+import de.rki.coronawarnapp.util.coroutine.DispatcherProvider
+import de.rki.coronawarnapp.util.flow.combine
+import de.rki.coronawarnapp.util.ui.SingleLiveEvent
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModel
+import de.rki.coronawarnapp.util.viewmodel.SimpleCWAViewModelFactory
+import timber.log.Timber
+
+class SubmissionTestResultNegativeViewModel @AssistedInject constructor(
+    dispatcherProvider: DispatcherProvider,
+    private val submissionRepository: SubmissionRepository
+) : CWAViewModel(dispatcherProvider = dispatcherProvider) {
+
+    val routeToScreen = SingleLiveEvent<NavDirections>()
+    val testResult: LiveData<TestResultUIState> = combine(
+        submissionRepository.deviceUIStateFlow,
+        submissionRepository.testResultReceivedDateFlow
+    ) { deviceUiState, resultDate ->
+        TestResultUIState(
+            deviceUiState = deviceUiState,
+            testResultReceivedDate = resultDate
+        )
+    }.asLiveData(context = dispatcherProvider.Default)
+
+    fun onBackPressed() {
+        routeToScreen.postValue(
+            SubmissionTestResultNegativeFragmentDirections
+                .actionSubmissionTestResultNegativeFragmentToHomeFragment()
+        )
+    }
+
+    fun deregisterTestFromDevice() {
```

Comment:
I saw a method for this in the repo

Reply:
Was merged in a different PR, I've changed the code to use that new method. :+1: